### PR TITLE
Generate clients as interfaces instead of structs

### PR DESCRIPTION
**Pull request by ChuntaoLu**
This PR generates the http/tchannel clients as client interfaces instead of structs in order to easily mock clients for testing.

### CODE REVIEW

**Comment by ChuntaoLu** [comment]
@Raynos here is another example where Travis is not triggered.

File: codegen/module.go [hunk 1]
```diff
@@ -503,7 +503,7 @@ func readPackageInfo(
 		),
 		ExportName:            "New" + qualifiedClassName,
 		QualifiedInstanceName: qualifiedInstanceName,
-		ExportType:            qualifiedInstanceName + qualifiedClassName,
+		ExportType:            qualifiedClassName,
```
Comment (Raynos) [inline codegen/module.go]: Why change this part ?

File: codegen/templates/dependency_struct.tmpl [hunk 2]
```diff
@@ -20,7 +20,7 @@ type Dependencies struct {
 // {{$classType | pascal}}Dependencies contains {{$classType}} dependencies
 type {{$classType | pascal}}Dependencies struct {
 	{{ range $idx, $dependency := $moduleInstances -}}
-	{{$dependency.PackageInfo.QualifiedInstanceName}} *{{$dependency.PackageInfo.ImportPackageAlias}}.{{$dependency.PackageInfo.ExportType}}
+	{{$dependency.PackageInfo.QualifiedInstanceName}} {{$dependency.PackageInfo.ImportPackageAlias}}.{{$dependency.PackageInfo.ExportType}}
```
Comment (Raynos) [inline codegen/templates/dependency_struct.tmpl]: not all dependencies are going to be interfaces. We will need something smarter for this.

File: codegen/templates/http_client.tmpl [hunk 3]
```diff
@@ -17,30 +17,55 @@ import (
...
 	baseURL := "http://" + ip + ":" + strconv.Itoa(int(port))
 	return &{{$clientName}}{
-		ClientID: "{{$clientID}}",
-		HTTPClient: zanzibar.NewHTTPClient(gateway, baseURL),
+		clientID: "{{$clientID}}",
+		httpClient: zanzibar.NewHTTPClient(gateway, baseURL),
 	}
 }
 
+// {{$clientName}} is the http client.
```
Comment (Raynos) [inline codegen/templates/http_client.tmpl]: nitpick: move the struct above the newX() func ?

File: codegen/module.go (hunk 1 above)
Comment (ChuntaoLu) [inline codegen/module.go]: The instance name is already (or should be) in the importing package name. Compare `bar.Client` with `bar.BarClient`, I think the former is cleaner and reads better, and linting for avoiding stutter also approves the former.

File: codegen/templates/dependency_struct.tmpl (hunk 2 above)
Comment (ChuntaoLu) [inline codegen/templates/dependency_struct.tmpl]: Good point, I will drop a TODO here

File: codegen/templates/http_client.tmpl (hunk 3 above)
Comment (ChuntaoLu) [inline codegen/templates/http_client.tmpl]: Done

**Pull request merged**